packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -57,6 +58,11 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
         ensureFluidResolvedUrl(createNewResolvedUrl);
 
         let odspResolvedUrl = getOdspResolvedUrl(createNewResolvedUrl);
+        const resolveUrlData: IOdspUrlParts = {
```

Comment:
Consider just passing odspResolvedUrl as-is rather than cloning.  It will pass the type check since TypeScript uses duck typing and will avoid the runtime impact of creating the new object.  I generally wouldn't bother cloning unless the function would have a side effect and you want to protect the caller's object.

Reply:
(and below as well)

Reply:
Hmmm maybe I take it back, I see now you're just spreading this object and passing to the host's token fetcher.  I still think it would be fine to just pass around an object with extra info, but won't push back if you want to tighten it up at runtime.

Reply:
If you leave it as-is, nit:  make sure all these vars are named `resolvedUrlData` not `resolveUrlData`.

Reply:
I had tried to keep the existing one and it seems to work too - I just thought it would be confusing and prone to error later on, to leave it as it was. I'm ok either way - will sync yo with you.